modules/hostap/src/supp_events.c

```diff
@@ -94,15 +101,17 @@ static int wpa_supp_process_status(struct supp_int_event_data *event_data, char
 	case WPA_SUPP_EVENT_BSS_ADDED:
 		ret = sscanf(wpa_supp_status + strlen("CTRL-EVENT-BSS-ADDED "), "%u "MACSTR,
 			&data->bss_added.id,
-			MAC2STR(data->bss_added.bssid));
+			MACADDR2STR(tmp_mac_addr));
+		os_memcpy(data->bss_added.bssid, tmp_mac_addr, ETH_ALEN);
 		event_data->data_len = sizeof(data->bss_added);
 		break;
 	case WPA_SUPP_EVENT_BSS_REMOVED:
 		ret = sscanf(wpa_supp_status + strlen("CTRL-EVENT-BSS-REMOVED "),
 			"%u "MACSTR,
 			&data->bss_removed.id,
-			MAC2STR(data->bss_removed.bssid));
+			MACADDR2STR(tmp_mac_addr));
 		event_data->data_len = sizeof(data->bss_removed);
+		os_memcpy(data->bss_removed.bssid, tmp_mac_addr, ETH_ALEN);
```

Comment:
The `tmp_mac_addr` is 4 * 6 = 24 bytes long array, but the `bss_removed.bssid` is 6 bytes long, the memcpy will copy garbage to bssid. You need to copy the fields one by one.